lmdk/cmake/xtensa-toolchain.cmake

```diff
@@ -0,0 +1,25 @@
+# Xtensa CMake toolchain file. Apply it using CMAKE_TOOLCHAIN_FILE variable.
+
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_VERSION 1)
+
+cmake_path(CONVERT $ENV{XTENSA_TOOLCHAIN_PATH} TO_CMAKE_PATH_LIST XTENSA_TOOLCHAIN_PATH)
```

Comment:
Started testing this PR and found few issues.
Issue 1: when XTENSA_TOOLCHAIN_PATH variable is not set, we got following error:

> CMake Error at C:/Workspace/my_sof_workspace/sof/lmdk/cmake/xtensa-toolchain.cmake:15 (cmake_path):
>   cmake_path CONVERT must be called with three or four arguments.

This is not very elegant, proper error message should be displayed.
Issue 2: toolchain is ignored and other installed is picked if anything fails in xtensa-toolchain.cmake script.
repro steps:
1. `cd sof\lmdk\libraries\dummy`
2. `set XTENSA_TOOLCHAIN_PATH=qweqew`
3. `cmake -S . -B build -G Ninja --toolchain "..\..\cmake\xtensa-toolchain.cmake" -DRIMAGE_COMMAND="qweqew" -DSIGNING_KEY="whateverkey"`
Result:
[cmake.log](https://github.com/thesofproject/sof/files/11390723/cmake.log)

As you can see, if we do not file any error message in the toolchain file, other installed toolchain on the end-user computer is taken (what is certainly not desired).